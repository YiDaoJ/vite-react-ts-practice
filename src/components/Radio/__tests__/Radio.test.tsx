import { render, screen, fireEvent } from "@testing-library/react";
import React from "react";
import userEvent from "@testing-library/user-event";
import { RadioButton as Radio } from "..";

describe("Radio test", () => {
  test("render Radio component", () => {
    const handleChange = jest.fn();
    const { asFragment } = render(
      <Radio label='Test' name='test' checked={false} onChange={handleChange} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("Radio change event", async () => {
    const handleChange = jest.fn();
    const user = userEvent.setup();

    render(<Radio label='Test' name='test' checked onChange={handleChange} />);

    // const input = screen.getByRole("input");
    //  Unable to find an accessible element with the role "input"
    // Qn: difference between userEvent and fireEvent

    const element = screen.getByLabelText("Test");
    // screen.debug(element);
    await user.click(element);
    expect(element).toBeChecked();
    // screen.debug(element);
  });

  test("disabled Radio", () => {
    const handleChange = jest.fn();
    const user = userEvent.setup();

    render(
      <Radio
        label='TestDisabled'
        name='test-disabled'
        onChange={handleChange}
        disabled
      />
    );

    const element = screen.getByLabelText("TestDisabled");

    // Qn:
    // await user.click(element);
    // // Error: Unable to perform pointer interaction as the element has `pointer-events: none`:
    // expect(elementF.checked).toEqual(false);

    // screen.debug(element);
    expect(element).toBeDisabled();
  });
});
