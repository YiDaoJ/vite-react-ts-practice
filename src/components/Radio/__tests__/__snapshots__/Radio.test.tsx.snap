// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio test render Radio component 1`] = `
<DocumentFragment>
  <label
    class="css-l0p67e-RadioWrapper examj981"
  >
    <span
      class="radio-input"
    >
      <input
        class="css-1ayomz5-StyledInput-inputActive examj980"
        name="test"
        type="radio"
      />
    </span>
    <span
      class="radio-label css-1wlzwar-labelStyle"
    >
      Test
    </span>
  </label>
</DocumentFragment>
`;
